internal/dataplane/parser/translators/ingress.go

```diff
@@ -153,7 +154,17 @@ func (i *ingressTranslationIndex) Add(ingress *netv1.Ingress, addRegexPrefix add
 
 func (i *ingressTranslationIndex) Translate() map[string]kongstate.Service {
 	kongStateServiceCache := make(map[string]kongstate.Service)
-	for _, meta := range i.cache {
+	// although this has no effect on the end route set, some parser tests are sensitive to route indices,
+	// so we sort here to avoid flakes
+	keys := make([]string, len(i.cache))
+	n := 0
+	for key := range i.cache {
+		keys[n] = key
+		n++
+	}
```

Comment:
I'm not sure if that wouldn't cause too many allocations for us (haven't measured it).

In general using a deterministic order of routes does make sense 👍 
Perhaps we could consider using something like https://pkg.go.dev/container/heap instead of a `map` in `ingressTranslationIndex.cache` which would always enforce the proper instead of needing to allocating a slice each time `Translate()` is called?

Reply:
Having to modify the items over rules to add new paths complicates making the whole thing a heap, but just the names is fine.